dalm/datasets/reading_comprehension_generation/synthetic_based.py

```diff
@@ -0,0 +1,187 @@
+            continue
+
+        if chunk:
+            for chunk_ in text_chunker(text, tokenizer, k):
+                gen_text = generate_synthetic_data(model_pipeline, chunk_, generation_params)
+                yield file, chunk_, gen_text
+        else:
+            gen_text = generate_synthetic_data(model_pipeline, text, generation_params)
+            yield file, text, gen_text
+
+def parse_args():
+    parser = argparse.ArgumentParser("Generate synthetic dataset for reading comprehension")
+    parser.add_argument("--model_name", type=str, default="HuggingFaceH4/zephyr-7b-alpha")
+    parser.add_argument("--input_directory", type=str, required=True, help="Directory containing the input files")
+    parser.add_argument(
+        "--output_directory",
+        type=str,
+        required=True,
+        help="Directory to save the generated files (serves as intermediate step and for debugging purposes)",
+    )
+    parser.add_argument(
+        "--state_file",
+        type=str,
+        required=False,
+        default="rc_generation_state.pkl",
+        help="File to save the state of the generation",
+    )
+    parser.add_argument("--context_length", type=int, default=4192, help="context length to calulcate the chunk size")
+    parser.add_argument("--no_chunk", action="store_false")
+    parser.add_argument(
+        "--dataset_name", type=str, default="synthetic_rc_dataset", help="name of the dataset to be saved"
+    )
+
+    return parser.parse_args()
+
+def main():
+    args = parse_args()
+    """
+    Pipeline here includes chunking, generation and parsing of question and answer into a list of exchanges
+    that can be used directly for training
+    """
+
+    if args.state_file:
+        if os.path.exists(args.state_file):
+            with open(args.state_file, "rb") as f:
+                state = pickle.load(f)
+        else:
+            state = {"processed_files": []}
+            pickle.dump(state, open(args.state_file, "wb"))
+
+    if not os.path.exists(args.output_directory):
+        os.makedirs(args.output_directory)
+
+    files_missed = 0
+    total_files = 0
+
+    synth_dataset_generator = generate_synthetic_dataset(
+        model_name=args.model_name,
+        input_directory=args.input_directory,
+        processed_files=state["processed_files"] if args.state_file else [],
+        chunk=args.no_chunk,
+        context_length=args.context_length,
+    )
+
+    for index, (filename, context, gen_text) in enumerate(synth_dataset_generator):
+        state["processed_files"].append(filename)
+        pickle.dump(state, open(args.state_file, "wb"))
+        qanda = question_and_answer_extractor(gen_text, context)
+        if qanda:
+            output_file = f"gen_{index}.json"
+            with open(os.path.join(args.output_directory, output_file), "w") as o:
+                json.dump(qanda, o)
+        else:
+            logger.warning(f"No question and answer pairs found for {filename}")
```

Comment:
This warning is useful, but it's slightly misleading when using chunking.  For example when the output is:

```
11/21/2023 16:42:54 - WARNING - root - No question and answer pairs found for context_data_66b189d4-f771-44da-a689-9837925ea4a7.csv
11/21/2023 16:43:11 - WARNING - root - No question and answer pairs found for context_data_66b189d4-f771-44da-a689-9837925ea4a7.csv
```

Could it list a chunk number or unique identifier to disambiguate?